pkgs/development/compilers/purescript/psc-package/default.nix

```diff
@@ -1,27 +1,56 @@
-{ haskellPackages, mkDerivation, fetchFromGitHub, lib }:
+# Copied from https://github.com/justinwoo/easy-purescript-nix/blob/master/psc-package-simple.nix
+{ stdenv, lib, fetchurl, gmp, zlib, libiconv, darwin }:
 
-with lib;
+let
+  dynamic-linker = stdenv.cc.bintools.dynamicLinker;
+
+in
+stdenv.mkDerivation rec {
+  pname = "psc-package-simple";
 
-mkDerivation rec {
-  pname = "psc-package";
   version = "0.6.2";
 
-  src = fetchFromGitHub {
-    owner = "purescript";
-    repo = pname;
-    rev = "v${version}";
-    sha256 = "0536mijma61khldnpbdviq2vvpfzzz7w8bxr59mvr19i10njdq0y";
+  src = if stdenv.isDarwin
+  then fetchurl {
+    url = "https://github.com/purescript/psc-package/releases/download/v0.6.2/macos.tar.gz";
+    sha256 = "17dh3bc5b6ahfyx0pi6n9qnrhsyi83qdynnca6k1kamxwjimpcq1";
+  }
+  else fetchurl {
+    url = "https://github.com/purescript/psc-package/releases/download/v0.6.2/linux64.tar.gz";
+    sha256 = "1zvay9q3xj6yd76w6qyb9la4jaj9zvpf4dp78xcznfqbnbhm1a54";
   };
 
-  isLibrary = false;
-  isExecutable = true;
+  buildInputs = [ gmp zlib ];
+
+  libPath = lib.makeLibraryPath buildInputs;
+
+  dontStrip = true;
+
+  installPhase = ''
+    mkdir -p $out/bin
 
-  executableHaskellDepends = with haskellPackages; [
-    aeson aeson-pretty either errors optparse-applicative
-    system-filepath turtle
-  ];
+    PSC_PACKAGE=$out/bin/psc-package
 
-  description = "A package manager for PureScript based on package sets";
-  license = licenses.bsd3;
-  maintainers = with lib.maintainers; [ Profpatsch ];
+    install -D -m555 -T psc-package $PSC_PACKAGE
+    chmod u+w $PSC_PACKAGE
+  '' + lib.optionalString stdenv.isDarwin ''
+    install_name_tool \
+      -change /usr/lib/libSystem.B.dylib ${darwin.Libsystem}/lib/libSystem.B.dylib \
+      -change /usr/lib/libiconv.2.dylib ${libiconv}/libiconv.2.dylib \
+      $PSC_PACKAGE
+  '' + lib.optionalString (!stdenv.isDarwin) ''
+    patchelf --interpreter ${dynamic-linker} --set-rpath ${libPath} $PSC_PACKAGE
+  '' + ''
+    chmod u-w $PSC_PACKAGE
+
+    mkdir -p $out/etc/bash_completion.d/
+    $PSC_PACKAGE --bash-completion-script $PSC_PACKAGE > $out/etc/bash_completion.d/psc-package-completion.bash
+  '';
+
+  meta = with lib; {
+    description = "A package manager for PureScript based on package sets";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ Profpatsch ];
+    platforms = [ "x86_64-darwin" "x86_64-linux" ];
```

Comment:
https://github.com/justinwoo/easy-purescript-nix/blob/master/psc-package-simple.nix did not contain a platforms declaration so I created this based on the releases. I'm not sure if there are any other platforms besides `"x86_64-linux" that work with the linux64 binary download.